src/components/SectionContainer.js

```diff
@@ -0,0 +1,346 @@
+           comments={this.props.sectionComments[mainSection.id] ?
+                     this.props.sectionComments[mainSection.id].data : []}
+           canComment={this.isMainSectionCommentable(hearing, user)}
+           onPostComment={this.onPostHearingComment.bind(this)}
+           canVote={this.isMainSectionVotable(user)}
+           onPostVote={this.onVoteComment.bind(this)}
+           canSetNickname={user === null}
+          />
+        </div>
+        <hr/>
+        <a href={reportUrl}><FormattedMessage id="downloadReport"/></a>
+      </div>
+    );
+  }
+
+  getQuestionLinksAndStuff(sectionGroups) {
+    const {hearing: {slug: hearingSlug}, section: {id: sectionId}} = this.props;
+    const questions = sectionGroups.reduce((questionsArray, currentSection) =>
+      [...questionsArray, ...currentSection.sections], []);
+    const currentIndex = questions.findIndex((question) => question.id === sectionId);
+    const prevPath = currentIndex !== 0 ? getSectionURL(hearingSlug, questions[currentIndex - 1]) : undefined;
+    const nextPath = currentIndex !== questions.length - 1 ? getSectionURL(hearingSlug, questions[currentIndex + 1]) : undefined;
+    return {
+      currentNum: currentIndex + 1,
+      totalNum: questions.length,
+      prevPath,
+      nextPath
+    };
+  }
+
+  render() {
+    const {hearing, section, user, sectionComments, language} = this.props;
+    // const hearingAllowsComments = acceptsComments(hearing);
+    // const closureInfoSection = this.getClosureInfo(hearing);
+    // const regularSections = hearing.sections.filter((section) => !isSpecialSectionType(section.type));
+    const mainSection = getMainSection(hearing);
+    const regularSections = hearing.sections.filter((sect) => !isSpecialSectionType(sect.type));
+    const sectionGroups = groupSections(regularSections);
+    const sectionNav = this.getQuestionLinksAndStuff(sectionGroups);
+    const isQuestionView = true;
+    // const fullscreenMapPlugin = hasFullscreenMapPlugin(hearing);
+    return (
+      <div className="section-container">
+        <LabelList className="main-labels" labels={hearing.labels}/>
+
+        <h1 className="page-title">
+          {this.getFollowButton()}
+          {!hearing.published ? <Icon name="eye-slash"/> : null}
+          {getAttr(hearing.title, language)}
+        </h1>
+
+        <Row>
+          <Sidebar currentlyViewed={'#hearing-sectiongroup'} isQuestionView={isQuestionView} activeSection={section} hearing={hearing} mainSection={mainSection} sectionGroups={sectionGroups}/>
+          <Col md={8} lg={9}>
+            <div className="section-browser">
+              <Link className="to-hearing" to={getHearingURL(hearing)}>
+                <Icon name="angle-double-left"/>&nbsp;<FormattedMessage id="hearing"/>
+              </Link>
+              <LinkWrapper disabled={!sectionNav.prevPath} className={`previous ${sectionNav.prevPath ? '' : 'disabled'}`} to={sectionNav.prevPath || '#'}>
```

Comment:
In the previous and next links, the section type name is selected based on the current section type even if we have sections of different types.

Here, sections 1-5 are called "Kysymys" and the last section is called "Hakulomake". Update the link texts depending on the section type.

Also, when we have several section types, only the first section type is listed in the sidebar. This should be fixed, though, when we display all sections in the sidebar regardless of where we are.
![image](https://cloud.githubusercontent.com/assets/9367712/21992797/86ed6df2-dc20-11e6-8f58-684f47c21cab.png)